src/main/java/it/cnr/isti/hpc/wikipedia/parser/ArticleParser.java

```diff
@@ -519,15 +521,17 @@ private void setLists(Article article, ParsedPage page) {
 	private void setDisambiguation(Article a) {
 
 		for (String disambiguation : locale.getDisambigutionIdentifiers()) {
-			if (StringUtils.containsIgnoreCase(a.getTitle(), disambiguation)) {
+			if (StringUtils.containsIgnoreCase(a.getTitle(), "(" + disambiguation + ")")) {
```

Comment:
In all fairness, it can contain anything - I've seen horrible things. But `(disambiguation)` is by far the most frequent. If some will slip through - it's fine, there's ultra tiny percentage of those if any. And they will be caught by template search. I hope.